finality-aleph/src/metrics.rs

```diff
@@ -71,11 +71,26 @@ pub(crate) enum Checkpoint {
 
 #[derive(Clone)]
 pub struct Metrics<H: Key> {
-    inner: Arc<Mutex<Inner<H>>>,
+    inner: Option<Arc<Mutex<Inner<H>>>>,
 }
 
 impl<H: Key> Metrics<H> {
-    pub fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+    pub fn register<F: FnOnce(PrometheusError)>(
+        registry: Option<&Registry>,
+        on_registration_failed: F,
```

Comment:
Rather that invoking closures on failure it is better to return Result

Reply:
Thanks, nice to know.